src/libraries/Native/Unix/System.Security.Cryptography.Native/opensslshim.c

```diff
@@ -150,11 +154,13 @@ void InitializeOpenSSLShim(void)
     fn##_ptr = (TYPEOF(fn))(dlsym(libssl, #fn));
 
 #define FALLBACK_FUNCTION(fn) \
-    if (!(fn##_ptr = (TYPEOF(fn))(dlsym(libssl, #fn)))) { fn##_ptr = (TYPEOF(fn))local_##fn; }
+    if (!(tmp_ptr = dlsym(libssl, #fn))) { tmp_ptr = (void*)local_##fn; } \
+    fn##_ptr = (TYPEOF(fn))tmp_ptr;
 
 #define RENAMED_FUNCTION(fn,oldfn) \
-    fn##_ptr = (TYPEOF(fn))(dlsym(libssl, #fn));\
-    if (!fn##_ptr && !(fn##_ptr = (TYPEOF(fn))(dlsym(libssl, #oldfn)))) { fprintf(stderr, "Cannot get required symbol " #oldfn " from libssl\n"); abort(); }
+    tmp_ptr = dlsym(libssl, #fn);\
+    if (!tmp_ptr && !(tmp_ptr = dlsym(libssl, #oldfn))) { fprintf(stderr, "Cannot get required symbol " #oldfn " from libssl\n"); abort(); } \
+    fn##_ptr = (TYPEOF(fn))tmp_ptr;
```

Comment:
I thought that in C I could write `fn##_ptr = (TYPEOF(fn))(dlsym(libssl, #fn) || dlsym(libssl, #oldfn))` to achieve the same thing, but the compiler disagreed.  So, I wrote it longhand.